Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/Inspector/ContentWidget.cpp

```diff
@@ -24,24 +24,57 @@ namespace EMStudio
         setLayout(m_layout);
     }
 
-    void ContentWidget::Update(const QString& headerTitle, const QString& iconFilename, QWidget* widget)
+    ContentWidget::~ContentWidget()
     {
+        CleanupOldWidget();
+    }
+
+    void ContentWidget::CleanupOldWidget()
+    {
+        // The content widget does not own the widget, thus remove it from the layout to not
+        // auto delete it along with destructing the content widget.
         if (m_widget)
         {
             m_widget->hide();
-            m_widget->deleteLater();
+            m_layout->removeWidget(m_widget);
+            m_widget->setParent(nullptr);
         }
+    }
+
+    void ContentWidget::Update(QWidget* widget)
+    {
+        UpdateInternal(/*headerTitle=*/{}, /*iconFilename=*/{}, widget, /*showHeader=*/false);
+    }
+
+    void ContentWidget::UpdateWithHeader(const QString& headerTitle, const QString& iconFilename, QWidget* widget)
+    {
+        UpdateInternal(headerTitle, iconFilename, widget, /*showHeader=*/true);
+    }
+
+    void ContentWidget::UpdateInternal(const QString& headerTitle, const QString& iconFilename, QWidget* widget, bool showHeader)
+    {
+        CleanupOldWidget();
         m_widget = widget;
 
         if (widget)
         {
             m_layout->addWidget(widget);
+            m_widget->show();
+        }
+
+        if (showHeader)
+        {
             m_headerWidget->Update(headerTitle, iconFilename);
+            m_headerWidget->show();
+        }
+        else
+        {
+            m_headerWidget->hide();
         }
     }
 
     void ContentWidget::Clear()
     {
-        Update(/*headerTitle=*/{}, /*iconFilename=*/{}, /*widget=*/nullptr);
+        UpdateWithHeader(/*headerTitle=*/{}, /*iconFilename=*/{}, /*widget=*/nullptr);
```

Comment:
Is this generally how parameters are commented throughout EMFX? What is the deciding factor on whether or not to comment the parameters?

Reply:
Yeah, that's the standard we use for it. Basically every time there is no variable involved where you can get the sense from its name. So by just throwing in {}, viewers have no clue what kind of object this is and then we add these comments.